utils/math/averager_heap.go

```diff
@@ -4,16 +4,13 @@
 package math
 
 import (
-	"container/heap"
-
 	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/heap"
 )
 
-var (
-	_ AveragerHeap   = averagerHeap{}
-	_ heap.Interface = (*averagerHeapBackend)(nil)
-)
+var _ AveragerHeap = averagerHeap{}
 
+// TODO replace this interface with utils/heap
```

Comment:
I don't want to do this in this PR because of coreth's dependency, I can do this as a follow-up.